### DISCUSSION TITLE
Analysing gas fees - why more in empty storage?

### DISCUSSION THREAD

**Discussion by codeitnav** [Cyfrin/foundry-full-course-cu#2505] [Q&A] [answered]
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.27;

contract SimpleStorage {

    uint256 public favNum = 88; 

    function store(uint256 _favNum) public {
        favNum = _favNum;
        retrieve();
    }

    function retrieve() public view returns(uint256){
        return favNum;
    }
}

on deploying on remix, 
gas fees = 180,610 gas --> for empty storage input
gas fees = 30,795 gas --> for storage input = 5

why so?

**Comment by EngrPips** [comment, marked as answer]
Hello @codeitnav, I am unsure if I correctly understand your question, but in solidity, it costs more the first time you want to store value in storage because it usually incurs additional warm-up costs. This article https://medium.com/coinmonks/solidity-storage-how-does-it-work-8354afde3eb did a great explanation of this concept.
